python/mirascope/llm/providers/openai/completions/provider.py

```diff
@@ -1,456 +1,71 @@
-"""OpenAI client implementation."""
+"""OpenAI Completions API provider implementation."""
 
-from collections.abc import Sequence
-from typing import Literal
-from typing_extensions import Unpack
+from collections.abc import Callable
 
-from openai import AsyncOpenAI, OpenAI
+from ...openai_compatible_provider import OpenAICompatibleProvider
+from ..model_id import model_name
 
-from ....context import Context, DepsT
-from ....formatting import Format, FormattableT
-from ....messages import Message
-from ....responses import (
-    AsyncContextResponse,
-    AsyncContextStreamResponse,
-    AsyncResponse,
-    AsyncStreamResponse,
-    ContextResponse,
-    ContextStreamResponse,
-    Response,
-    StreamResponse,
-)
-from ....tools import (
-    AsyncContextTool,
-    AsyncContextToolkit,
-    AsyncTool,
-    AsyncToolkit,
-    ContextTool,
-    ContextToolkit,
-    Tool,
-    Toolkit,
-)
-from ...base import BaseProvider, Params
-from ..model_id import OpenAIModelId, model_name
-from . import _utils
 
-
-class OpenAICompletionsProvider(BaseProvider[OpenAI]):
-    """The client for the OpenAI LLM model."""
+class OpenAICompletionsProvider(OpenAICompatibleProvider):
+    """Provider for OpenAI's ChatCompletions API."""
 
     id = "openai:completions"
     default_scope = "openai/"
+    default_base_url = None
+    api_key_env_var = "OPENAI_API_KEY"
+    api_key_required = False
+    provider_name = "OpenAI"
 
     def __init__(
         self,
         *,
         api_key: str | None = None,
         base_url: str | None = None,
         wrapped_by_openai_provider: bool = False,
```

Comment:
I think we can remove wrapped_by_openai_provider in favor of the effective_provider_id now